src/components/Navigation/Navigation.js

```diff
@@ -0,0 +1,67 @@
+import { useState } from 'react'
+import { Link } from 'react-router-dom'
+import NavButton from './NavButton'
+import styled from 'styled-components/macro'
+import { NavData } from './NavData'
+
+function Navigation() {
+  const [navigation, setNavigation] = useState(false)
+
+  function showNavigation() {
+    setNavigation(!navigation)
+  }
+
+  return (
+    <>
+      <NavButton showNavigation={showNavigation} navigation={navigation} />
+      <Nav navigation={navigation}>
+        <NavButton showNavigation={showNavigation} navigation={navigation} />
+        {NavData.map((item, index) => {
+          return (
+            <NavItem key={index}>
+              <Link to={item.path}>
+                {item.icon}
+                <span>{item.title}</span>
+              </Link>
+            </NavItem>
+          )
+        })}
+      </Nav>
+    </>
+  )
+}
+
+const Nav = styled.nav`
+  background: #c9ccd3;
+  background-image: linear-gradient(
+    -180deg,
+    rgba(255, 255, 255, 0.5) 0%,
+    rgba(0, 0, 0, 0.5) 100%
+  );
+  background-blend-mode: lighten;
+  background-blend-mode: multiply, multiply;
+  width: 50%;
+  height: 100vh;
+  display: flex;
+  flex-direction: column;
+  position: fixed;
+  top: 0;
+  right: -100%;
```

Comment:
Right now, clicking on the menu button ***visually*** doesn't move in the menu right away. Because your `nav` has the property `width: 50%`, but its initial position is `right: -100%` half of your animation time is wasted and it feels like a 0.425s delay (half of `transition: 850ms`). You should set its initial position to match its total width:

```suggestion
  right: -50%;
```